### PR TITLE
Code cleanup and contract version support

### PR DESCRIPTION
**Pull request by kameshkishoreIonixx**
## Related Issue
 <JIRA_ID with description>

## Changes

1. <#>
2. <#>

## Checklist

- [ ] param and service should be meaningful(name should indicate, why it is used rather than which service its calling.)
- [ ] Param should be type and description in ReadMe
- [ ] don't use arrow function functional components useCallback
- [ ] Always try to use const. There is a very good reason to use let
- [ ] Should be exported only Hooks and interface
- [ ] Before pushing to GitHub delete all unused states, spacing, and console.log, if exist
- [ ] PR should contain  screenshots Unit test with 100% coverage
- [ ] Lint should be passed on to PR

### CODE REVIEW

File: example/src/layout/CheckoutLayout.tsx [hunk 1]
```diff
@@ -57,12 +57,12 @@ const expressOptions: CheckboxOptions[] = [
   { field: 'expressMetamask', label: 'MetaMask', checked: true },
 ];
 const paymentOptions: CheckboxOptions[] = [
-  { field: 'gpay', label: 'Gpay', checked: true },
-  { field: 'applepay', label: 'Applepay', checked: true },
-  { field: 'walletconnect', label: 'WalletConnect', checked: true },
+  // { field: 'gpay', label: 'Gpay', checked: true },
+  // { field: 'applepay', label: 'Applepay', checked: true },
```
Comment (sakthi-ionixx) [inline example/src/layout/CheckoutLayout.tsx:61]: Remove commented code

File: example/src/lib/providers/PaymentProvider.tsx [hunk 2]
```diff
@@ -392,31 +392,51 @@ export const PaymentProvider = ({
...
+            [''],
+            collectionData?.supportedCurrency?.contractAddress,
+            utils.parseEther(value?.toString()),
+            ethers.constants.AddressZero,
+            '0',
+          ];
+        } else {
+          tokenDetails = [
+            collectionData?.saleId || collectionData?.collectionId,
+            paymentReceipt.invoiceDetails?.items[0]?.onChainPaymentInfo?.ownerWalletAddress,
```
Comment (sakthi-ionixx) [inline example/src/lib/providers/PaymentProvider.tsx:429]: Can you add ?  For undefined check

File: example/src/lib/views/index.tsx [hunk 3]
```diff
@@ -198,34 +184,32 @@ const MojitoCheckoutLayout = ({
...
-            onCloseModal={ onCloseWalletModal } />
-        </ThemeProvider>
-      ) }
+      <ThemeProvider theme={ uiConfiguration?.walletConnect?.theme || connectWalletTheme }>
+        <ConnectWalletContainer
+          open={ openWalletModal }
+          config={{
+            organizationId: uiConfiguration?.walletConnect?.orgId ?? '',
+            paperClientId: uiConfiguration?.walletConnect?.paperClientId ?? '',
+            paperNetworkName: getNetworkName(Number(network?.chainID)),
```
Comment (sakthi-ionixx) [inline example/src/lib/views/index.tsx:193]: Add undefined check

File: example/src/layout/CheckoutLayout.tsx (hunk 1 above)
Comment (kameshkishoreIonixx) [reply example/src/layout/CheckoutLayout.tsx:61]: the code was commented in outside the lib folder, we need this in upcoming feature

File: example/src/lib/providers/PaymentProvider.tsx (hunk 2 above)
Comment (kameshkishoreIonixx) [reply example/src/lib/providers/PaymentProvider.tsx:429]: i tried, we are passing data as undefined in contract it was not accepting undefined values while execute.

File: example/src/lib/views/index.tsx (hunk 3 above)
Comment (kameshkishoreIonixx) [reply example/src/lib/views/index.tsx:193]: the function will not return undefined, because it was hard code values.

**Review by sakthi-ionixx — APPROVED** [review]

**Review by sakthi-ionixx — APPROVED** [review]

**Pull request merged**